app/src/main/java/com/cmput301w23t09/qrhunter/qrcode/QRCode.java

```diff
@@ -65,22 +64,37 @@ public QRCode(String hash) throws ExecutionException, InterruptedException {
   public QRCode(
       String hash,
       String name,
-      Bitmap visualRepresentation,
       Integer score,
       Location loc,
       ArrayList<Photo> photos,
       ArrayList<Comment> comments,
-      ArrayList<String> players) {
+      ArrayList<String> players)
+      throws ExecutionException, InterruptedException {
     this.hash = hash;
     this.name = name;
-    this.visualRepresentation = visualRepresentation;
     this.score = score;
+    this.visualRepresentation =
+        new QRCodeVisualFetcher(this)
+            .execute("https://api.dicebear.com/5.x/pixel-art-neutral/jpg?seed=" + hash)
+            .get();
     this.loc = loc;
     this.photos = photos;
     this.comments = comments;
     this.players = players;
   }
 
+  /** This constructs a blank QR code with all fields set to null For testing purposes only */
+  public QRCode() {
```

Comment:
I feel that this constructor and the added `set` methods are a workaround to QRCode tests.

As a QRCode's name and score are determined by the hash. By exposing `setName` and `setScore`, it no longer makes the name or score reliant on the hash. I feel that `setName` and `setScore` should not exist in this case.